crates/snapshot/src/snapshotter.rs

```diff
@@ -91,308 +65,129 @@ impl<DB: Database> Snapshotter<DB> {
     /// Creates a new [Snapshotter].
     pub fn new(
         provider_factory: ProviderFactory<DB>,
-        snapshots_path: impl AsRef<Path>,
-        block_interval: u64,
-    ) -> RethResult<Self> {
-        let (highest_snapshots_notifier, highest_snapshots_tracker) = watch::channel(None);
-
-        let mut snapshotter = Self {
-            provider_factory,
-            snapshots_path: snapshots_path.as_ref().into(),
-            highest_snapshots: HighestSnapshots::default(),
-            highest_snapshots_notifier,
-            highest_snapshots_tracker,
-            block_interval,
-        };
-
-        snapshotter.create_directory()?;
-        snapshotter.update_highest_snapshots_tracker()?;
-
-        Ok(snapshotter)
+        snapshot_provider: Arc<SnapshotProvider>,
```

Comment:
I'm not sure I like this.

This types wrapps the snapshotprovider and the providerfactor, which itself also has a snapshotprovider.
could this be an issue?

Reply:
for `Snapshotter`, having a `SnapshotProvider` is a requirement, so we pass it on initialization. Yes, `ProviderFactory` also contains `SnapshotProvider`, but it returns `Option` because it may not be initialized, and we need to handle that. Just feels less ugly to have a non-optional `SnapshotProvider`.

regarding safety, I don't think it's a problem because it's already an `Arc` in the `ProviderFactory`.